marshmallow/schema.py

```diff
@@ -195,7 +195,7 @@ def __init__(self, meta):
         self.exclude = getattr(meta, 'exclude', ())
         if not isinstance(self.exclude, (list, tuple)):
             raise ValueError("`exclude` must be a list or tuple.")
-        self.strict = getattr(meta, 'strict', False)
+        self.strict = getattr(meta, 'strict', None)
```

Comment:
This prevents warnings on top level schemas also. Unless `strict` is explicitly set to `False`. I suspect the majority of users are relying on the default value, so this would have the same effect as removing the warning for most users.

Reply:
As a side note, I'm not sure the warning is _that_ useful. There are tons of changes between 2 and 3 anyway, we can't warn for everything. If the last resort solution is to remove the warning, that's fine IMHO. If you find an elegant solution to sort this out and keep the warning, that's fine with me as well.

Reply:
Yeah, I thought about just removing this particular warning, since it can add a fair amount of noise (I ended up [silencing it in flask-resty's tests](https://github.com/4Catalyzer/flask-resty/pull/232)). The change in `strict`  behavior is arguably the most significant change from 2 to 3 and is hard to miss when upgrading. Setting `strict=True` will not suffice, since you need to account for the change in return value type as well.

Anyway, I think this PR isn't the way to go for the reason @deckar01 pointed out, so closing.

So question is:  does this warning provide a significantly positive nudge to upgrade to marsmallow 3, or do the do the false positives produce too much noise to be useful? 🤔